google-cloud-tools-plugin/src/com/google/cloud/tools/intellij/appengine/sdk/CloudSdkService.java

```diff
@@ -39,6 +40,8 @@ public static CloudSdkService getInstance() {
 
   public abstract void setSdkHomePath(String path);
 
+  public abstract Set<CloudSdkValidationResult> validateCloudSdk(@NotNull Path pathToCloudSdk);
```

Comment:
I think we could use validations like this at both levels. For example, checking the sdk version is really specific to the intellij plugin. It would be nice to pull some of this validation into the common library in an extensible way.